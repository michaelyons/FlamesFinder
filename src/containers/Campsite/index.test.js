import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps, Campsite } from './index';
import {
  addCurrentWeather,
  addTenHourWeather,
  addTenDayWeather
} from '../../actions/weatherActions';
import {
  currentWeatherMock,
  facilityIDMock,
  contractIDMock,
  tenDayWeatherMock,
  tenHourWeatherMock,
  campObjectMock,
  mockText
} from './mockCampsiteData';

jest.mock('../../helper/apiCalls');

describe('Campsite Container', () => {
  beforeEach(async () => {
    campObjectMock;
    mockText;
  });

  describe('mapStateToProps', () => {
    it('should return a current object, a tenHour array, and a tenDay array', () => {
      const mockState = {
        currentWeather: {
          location: 'Denver'
        },
        tenHourWeather: [
          {
            time: '8:00 AM',
            temp: '64°F',
            feelsLike: '64°F',
            averageHumidity: '29%',
            condition: 'Clear',
            averageWind: 'WSW 7',
            uvIndex: '0/10',
            icon_url: 'http://icons.wxug.com/i/c/k/clear.gif'
          }
        ],
        tenDayWeather: [
          {
            day: 'Wednesday',
            date: '9/12/2018',
            high: '92°F',
            low: '62°F',
            icon: 'http://icons.wxug.com/i/c/k/clear.gif',
            conditions: 'Clear',
            averageWind: 'S 8 MPH'
          }
        ]
      };

      const expected = {
        currentWeather: {
          location: 'Denver'
        },
        tenHourWeather: [
          {
            time: '8:00 AM',
            temp: '64°F',
            feelsLike: '64°F',
            averageHumidity: '29%',
            condition: 'Clear',
            averageWind: 'WSW 7',
            uvIndex: '0/10',
            icon_url: 'http://icons.wxug.com/i/c/k/clear.gif'
          }
        ],
        tenDayWeather: [
          {
            day: 'Wednesday',
            date: '9/12/2018',
            high: '92°F',
            low: '62°F',
            icon: 'http://icons.wxug.com/i/c/k/clear.gif',
            conditions: 'Clear',
            averageWind: 'S 8 MPH'
          }
        ]
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
  describe('mapDispatchToProps', () => {
    it('should call dispatch with an addCurrentWeather action when addCurrentWeather is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addCurrentWeather({ location: 'Denver' });

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addCurrentWeather({ location: 'Denver' });

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
    it('should call dispatch with an addTenHourWeather action when addTenHourWeather is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addTenHourWeather([{ time: '9:00PM' }]);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addTenHourWeather([{ time: '9:00PM' }]);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
    it('should call dispatch with an addTenDayWeather action when addTenDayWeather is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addTenDayWeather([{ day: 'Friday' }]);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addTenDayWeather([{ day: 'Friday' }]);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
