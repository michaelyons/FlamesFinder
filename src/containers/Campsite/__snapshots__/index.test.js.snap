// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campsite Container should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Campsite />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h2>
                    Campsite Info
          </h2>
          <h3 />
          <p />
          <section>
                    <div>
                              Loading...
                    </div>
          </section>
          <p />
</div>,
        <div>
          <WeatherCard
                    currentWeather={undefined}
                    tenDayWeather={undefined}
                    tenHourWeather={undefined}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Campsite Info
</h2>,
            <h3 />,
            <p />,
            <section>
              <div>
                            Loading...
              </div>
</section>,
            <p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Campsite Info",
            },
            "ref": null,
            "rendered": "Campsite Info",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                Loading...
</div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading...",
              },
              "ref": null,
              "rendered": "Loading...",
              "type": "div",
            },
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WeatherCard
            currentWeather={undefined}
            tenDayWeather={undefined}
            tenHourWeather={undefined}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currentWeather": undefined,
            "tenDayWeather": undefined,
            "tenHourWeather": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h2>
                        Campsite Info
            </h2>
            <h3 />
            <p />
            <section>
                        <div>
                                    Loading...
                        </div>
            </section>
            <p />
</div>,
          <div>
            <WeatherCard
                        currentWeather={undefined}
                        tenDayWeather={undefined}
                        tenHourWeather={undefined}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Campsite Info
</h2>,
              <h3 />,
              <p />,
              <section>
                <div>
                                Loading...
                </div>
</section>,
              <p />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Campsite Info",
              },
              "ref": null,
              "rendered": "Campsite Info",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  Loading...
</div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Loading...",
                },
                "ref": null,
                "rendered": "Loading...",
                "type": "div",
              },
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WeatherCard
              currentWeather={undefined}
              tenDayWeather={undefined}
              tenHourWeather={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "currentWeather": undefined,
              "tenDayWeather": undefined,
              "tenHourWeather": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
